src/evict.c

```diff
@@ -512,7 +512,7 @@ static unsigned long evictionTimeLimitUs() {
 int performEvictions(void) {
     if (!isSafeToPerformEvictions()) return EVICT_OK;
 
-    int keys_freed = 0;
+    long keys_freed = 0;
```

Comment:
unsigned long? I suppose that is the max number of keys we support.

Reply:
I have simulated a real environment on my own computer, when the number of keys reaches 200 million, my 32g of memory should be half occupied, do we need 2^64 times as much.
I'm confused about the max keys the dict can support.

Reply:
If unsigned long is the max number of keys for a single db, then unsigned long long is not enough to store the number of keys for all db.

Reply:
I think there was another issue we were talking about this right? Redis officially supports 2^32 keys, and we were going to ignore that and make it support a long amount of keys. 

I think a hash table can support an "unsigned long" amount of data, so that seems to be the target we should aim for? It seems like everything, #key related should be an unsigned long.

Reply:
Please remember Redis can also run as a 32-bit process, so long could be as little as 2^32.

Reply:
> Please remember Redis can also run as a 32-bit process, so long could be as little as 2^32.

Yeah, long has been modified to unsigned.